### PR TITLE
Javascrip with objects

### PR DESCRIPTION
**Pull request by Obote**
What does this PR do?

- Create a collection that keeps a list of books
- Create a function to add a new book to the collection, with title and author.
- Create a function to remove a book from the collection 
- Display all books saved in the collection at the top part of the page.
- Make sure that data is preserved in the browser's memory by using localStorage

### CODE REVIEW

File: javascript/index.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  document.getElementById('title').value = '';
+  document.getElementById('author').value = '';
+};
+
+DisplayBook.prototype.deleteBook = function (bookTitle, bookAuthor) {
+  // eslint-disable-next-line max-len
+  this.books = this.books.filter((book) => !(book.bookTitle === bookTitle && book.bookAuthor === bookAuthor));
+  this.saveBooksToLocalStorage();
+};
+
```
Comment (DeliceLydia) [inline javascript/index.js]: - [ ] Please make sure all books are being stored in the `local storage` currently if I add the second book the previous one disappears
Screenshot 2023-07-05 at 10 52 06

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi Team, 

Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- All team members participated
- No linter errors


## Required Changes ♻️
_Check the comments under the review._



### Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: javascript/index.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+DisplayBook.prototype.saveBooksToLocalStorage = function () {
+  localStorage.setItem('books', JSON.stringify(this.books));
+};
```
Comment (wayungi) [inline javascript/index.js]: - [ ] Your local storage cannot keep more that one book. every time a book is added, the old one is deleted. Its my suggestion that you add the book at the books array and then you save the array to local storage. 😓 
redo

- [ OPTIONAL] What do you think about using function programming to implement this. 🤔

File: javascript/index.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+DisplayBook.prototype.clearForm = function () {
+  document.getElementById('title').value = '';
+  document.getElementById('author').value = '';
+};
+
+DisplayBook.prototype.deleteBook = function (bookTitle, bookAuthor) {
+  // eslint-disable-next-line max-len
+  this.books = this.books.filter((book) => !(book.bookTitle === bookTitle && book.bookAuthor === bookAuthor));
+  this.saveBooksToLocalStorage();
+};
```
Comment (wayungi) [inline javascript/index.js]: - [ ] the delete function is also deleting the wrong book. The  oldest book on the UI is the one that gets deleted. Please add an id field to the books and keep in unique then use this Id field to target the book to be deleted 👍🏾

**Review by wayungi — CHANGES_REQUESTED** [review]
Hi @Obote , 

image

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.


 ## Highlights
- [ ] All linter checks passing ✔️ 
- [ ]  Correct GitHub flow ✔️ 
- [ ] Descriptive PR body 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: javascript/index.js (hunk 3 above)
Comment (Obote) [reply javascript/index.js]: Hello @wayungi 
Thank you for taking the time to review our work. However, I wanted to bring to your attention that the specific code you mentioned does not exist anymore. Based on the feedback from the previous review, we made some revisions and updates which might have resulted in the removal or renaming of certain sections of the code. Therefore, it is possible that you may have made a mistake during the review. If you could kindly provide us with further details or clarify your concerns, we would be more than happy to address them accordingly.

**Review by Reem-lab — APPROVED** [review]
# Hi Team,  👋🏻👋🏻

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights

- [ ] No Linters errors 🔥
- [ ]  Great job working on the functionality💯
- [ ]  Followed Gitflow 🔥
- [ ]  Kudos for making the requested changes 🤩🚀

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**